Calendar/Calendar.js

```diff
@@ -1,3 +1,211 @@
+import Spottable from '@enact/spotlight/Spottable';
+import {Cell, Row} from '@enact/ui/Layout';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+
+import BodyText from '../BodyText';
+import Button, {ButtonBase} from '../Button';
+import Dropdown from '../Dropdown';
+import Skinnable from '../Skinnable';
+
+import componentCss from './Calendar.module.less';
+
+const SpottableButton = Spottable(ButtonBase);
+
+/**
+ * The calendar base component which sets-up the component's structure.
+ *
+ * This component is most often not used directly but may be composed within another component as it
+ * is within {@link sandstone/Calendar|Calendar}.
+ *
+ * @class CalendarBase
+ * @memberof sandstone/Calendar
+ * @ui
+ * @private
+ */
+
+const CalendarBase = kind({
+	name: 'Calendar',
+
+	functional: true,
+
+	propTypes: /** @lends sandstone/Calendar.CalendarBase.prototype */ {
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal elements and states of this component.
+		 *
+		 * The following classes are supported:
+		 *
+		 * `calendar` - The root class name
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		css: PropTypes.object,
+
+		/**
+		 * The `day` component of the Date.
+		 *
+		 * @type {Number}
+		 * @rivate
+		 */
+		day: PropTypes.number,
+
+		/**
+		 * Applies the `disabled` class.
+		 *
+		 * When `true`, the calendar is shown as disabled.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @private
+		 */
+		disabled: PropTypes.bool,
+
+		/**
+		 * The `month` component of the Date.
+		 *
+		 * @type {String}
+		 * @private
+		 */
+		month: PropTypes.string,
+
+		/**
+		 * The `year` component of the Date.
+		 *
+		 * @type {Number}
+		 * @private
+		 */
+		year: PropTypes.number
+	},
+
+	styles: {
+		css: componentCss,
+		publicClassNames: true
+	},
+
+	render: ({css, disabled = false}) => {
+		const DAYS = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+		const DAYS_LEAP = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+		const DAYS_OF_THE_WEEK = ['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN'];
+		const MONTHS = [
+			'January',
+			'February',
+			'March',
+			'April',
+			'May',
+			'June',
+			'July',
+			'August',
+			'September',
+			'October',
+			'November',
+			'December'
+		];
+		const YEARS = ['2020', '2021', '2022', '2023', '2024', '2025', '2026', '2027', '2028'];
+		const today = new Date();
+
+		const month = today.getMonth();
+		const year = today.getFullYear();
+		const yearIndex = YEARS.indexOf(year.toString());
+
+		const getStartDayOfMonth = () => {
+			const startDate = new Date(today.getFullYear(), today.getMonth(), 1).getDay();
+			return startDate === 0 ? 7 : startDate;
+		};
+
+		const startDay = getStartDayOfMonth(today);
+
+		const isLeapYear = () => {
+			return (year % 4 === 0 && year % 100 !== 0) || year % 400 === 0;
+		};
+
+		const days = isLeapYear(year) ? DAYS_LEAP : DAYS;
```

Comment:
after migrating to functional component, is it possible to all/some of these util functions to a separate file?